### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by vireshk**
Minor cleanups

### CODE REVIEW

**Comment by stefano-garzarella** [comment]
> @stefano-garzarella any idea why this is failing for the vsock test ? I haven't changed anything significant here.

ehm, no. In `VhostUserVsockBackend::new()` the only things that might fail are the creation of the UDS listener, the EventFD, and EpollFD.

IIUC, the tests run in threads of the same process.
Could it happen that we reach the limit of fd created?
Can we limit the parallelism?

**Comment by stefano-garzarella** [comment]
I replicated on my laptop running `while true; do cargo test -- --test-threads=2 || break; done`.
Using `--test-threads=1` never happens, using a value greater than 1 instead fails after a while and almost always with the creation of UDS or EpollFD or on epoll_ctl.
Strange, any ideas?

**Comment by vireshk** [comment]
> I replicated on my laptop running `while true; do cargo test -- --test-threads=2 || break; done`. Using `--test-threads=1` never happens, using a value greater than 1 instead fails after a while and almost always with the creation of UDS or EpollFD or on epoll_ctl. Strange, any ideas?

One guess I had was clashing file names, since you have fixed them in tests. And so we won't be able to create same files twice. I tried to get around it Using TempFiles, but couldn't succeed yet. Just pushed the branch to get it merged, if I can skip the error just once :)

This error is already there in main and needs to be fixed.

**Comment by stefano-garzarella** [comment]
> > I replicated on my laptop running `while true; do cargo test -- --test-threads=2 || break; done`. Using `--test-threads=1` never happens, using a value greater than 1 instead fails after a while and almost always with the creation of UDS or EpollFD or on epoll_ctl. Strange, any ideas?
> 
> One guess I had was clashing file names, since you have fixed them in tests. And so we won't be able to create same files twice. I tried to get around it Using TempFiles, but couldn't succeed yet. Just pushed the branch to get it merged, if I can skip the error just once :)

Can you confirm that using `--test-threads=1` it works?
> 
> This error is already there in main and needs to be fixed.

What I understand is that there is something wrong with fd when there are multiple test threads (not sure if related to O_CLOEXEC).

What we can do in the short term is to serialize vsock tests using `serial_test`, but only as a workaround to avoid test failures, then I'll try to figure out what is happening. I'll send that PR, so we don't block other PRs.

**Comment by vireshk** [comment]
Yay, the tests passed once at least :)

Lets get this merged now @mathieupoirier @stsquad .

I have already raised an issue for test failure though: https://github.com/rust-vmm/vhost-device/issues/232

**Comment by vireshk** [comment]
> Can you confirm that using `--test-threads=1` it works?

It hasn't failed for me for count 2 as well, 3 is minimum. But then I am only running vsock tests.

**Comment by vireshk** [comment]
> > Can you confirm that using `--test-threads=1` it works?
> 
> It hasn't failed for me for count 2 as well, 3 is minimum. But then I am only running vsock tests.

it failed for count 2 just now :)

**Comment by stefano-garzarella** [comment]
> > > Can you confirm that using `--test-threads=1` it works?
> > 
> > 
> > It hasn't failed for me for count 2 as well, 3 is minimum. But then I am only running vsock tests.
> 
> it failed for count 2 just now :)

Yep, also for me. Instead, using  `serial_test` seems to be fine (20 threads). As a workaround, we could merge it, and then I'll try to figure out what is happening.

**Review by stsquad — DISMISSED** [review]
LGTM! :thumbsup:

**Comment by vireshk** [comment]
@stefano-garzarella can you please approve this ?

File: crates/vsock/src/main.rs
Comment (stefano-garzarella) [inline crates/vsock/src/main.rs:31]: I was sure I solved it, I must have missed the commit while rebasing...

**Review by stefano-garzarella — APPROVED** [review]

**Comment by stefano-garzarella** [comment]
> @stefano-garzarella can you please approve this ?

Done, but it seems that the changes to CODEOWNERS are not effective yet, I don't know if I need to do anything else.

**Comment by stefano-garzarella** [comment]
> > @stefano-garzarella can you please approve this ?
> 
> Done, but it seems that the changes to CODEOWNERS are not effective yet, I don't know if I need to do anything else.

If I open CODEOWNERS this error message is printed:  _Unknown owner on line 2: make sure @stefano-garzarella exists and has write access to the repository_ 

Can you give me write access?

**Comment by vireshk** [comment]
> > > @stefano-garzarella can you please approve this ?
> > 
> > 
> > Done, but it seems that the changes to CODEOWNERS are not effective yet, I don't know if I need to do anything else.
> 
> If I open CODEOWNERS this error message is printed: _Unknown owner on line 2: make sure @stefano-garzarella exists and has write access to the repository_
> 
> Can you give me write access?

**Comment by stefano-garzarella** [comment]
> > > Can you give me write access?
> 
> Any idea how ?

Ehm no -.-, @lauralt @andreeaflorescu should know :-)

**Review by stefano-garzarella — APPROVED** [review]

**Comment by andreeaflorescu** [comment]
> > > > Can you give me write access?
> > 
> > 
> > Any idea how ?
> 
> Ehm no -.-, @lauralt @andreeaflorescu should know :-)

I just noticed the conversation here, I also commented on the initial PR where you were added as a codeowner. The changes are not automatic. The CODEOWNERS file is something we use for transparency so that people that submit PRs know who has permissions to merge their code. To get actual approval rights one of the existing maintainers has to add you to the repo. I've added you for now, feel free to use the documented process in the future.

**Pull request merged**
